### ISSUE TITLE
Add entity aliases

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#7324]
I suggest to add the following function to the Lua API:

`minetest.register_entity_alias(old_name, new_name)`

After this function is called, all entities in the game with the entitystring of `old_name` will be treated as if the really had the entitystring of `new_name`.

Basically, this function would work like `minetest.register_alias`, except it only aliases entitystrings instead of itemstrings.

**Comment by Wuzzy2** [comment]
Please reconsider this feature request. I think having entity aliases will be incredibly useful, especially if you need to rename legacy mods. This is not just a "nice to have" imo.

Note that entitystrings *must* begin with the modname and a colon, other names are forbidden. So if you must rename a mod `stars` and it registers an entity `stars:entity`, and then you must rename the mod to `planets`, you also MUST rename the entity to `planets:entity` but you can NOT alias it. Therefore, players will likely see unknown objects after an update, which is bad development practice.

An entity alias would prevent this.

**Comment by sfan5** [comment]
> Note that entitystrings _must_ begin with the modname and a colon, other names are forbidden.

Then why not just fix this?

**Comment by rubenwardy** [comment]
This seems fairly easy to implement, could just do a __newindex on the table. Although, that wouldn't be the most consistent I guess.

Entities are prototype based, and don't need to be registered with the c++ or at runtime, I'd be surprised if the convention is strictly enforced. But even if not, it's still a good idea to match the modname

**Comment by Wuzzy2** [comment]
Yeah, I agree with rubenwardy. It’s reasonable to have a `modname:entityname` naming rule for entities, please don’t drop this. It keeps things cleaner.

**Comment by sfan5** [comment]
I'm not talking about the naming rule in general. Weren't you saying that the `:foo:bar` override does not work with entities?

**Comment by Wuzzy2** [comment]
According to my tests, the override does indeed not work for entities.
